src/pages/EntityDetails/Model/Model.js

```diff
@@ -78,6 +82,26 @@ const Model = () => {
     activeView: withDefault(StringParam, "apps"),
   });
 
+  const uuid = modelStatusData?.info?.uuid;
+
+  useEffect(() => {
+    let watcherHandle = null;
+    let conn = null;
+    async function startWatcher() {
+      ({ conn, watcherHandle } = await startModelWatcher(uuid, appState));
```

Comment:
Is it possible that appState might not have the data you need at this point? I know there's a comment below about fixing this up, but if you end up with outdated data sent to `startModelWatcher` then it'll be because this useEffect was run before `appState` had all the data.

Reply:
It's not clear here but for a `uuid` to be defined then the necessary in `appState` will be populated. This is opaque and the only way to know is to "know" :) But yeah as per the following comment this needs to be updated. Thanks for calling it out 👍